specification/0051-qsl-versioning.md

```diff
@@ -0,0 +1,70 @@
+# Quilt Standard Libraries Versioning
+
+## Summary
+
+This document intends to specify the versioning scheme of the Quilt Standard Libraries to ensure consistent versioning, reduce confusion and address the case of experimental and beta versions.
+
+## Motivation
+
+As <abbr title="Quilt Standard Libraries">QSL</abbr> gets closer to a Beta release,
+it is important for it to figure out a versioning scheme.
+
+The versioning scheme will allow to ensure consistent version names, which reduce confusion among modders,
+and would make version comparison more robust.
+
+It is also important as it's a leading step in deciding how <abbr title="Quilt Standard Libraries">QSL</abbr> experimental or beta modules should be handled.
+
+Up until now QSL has been lightly versioned as it was in active development without release, taking the form of `1.0.0+<mc_version>-SNAPSHOT`. This was only temporary.
+
+## Explanation
+
+First of all, the versioning scheme is based of [semver] since it's mendatory per [the <abbr title="quilt.mod.json">QMJ</abbr> mod manifest specification][QMJ].
+
+This answers already a lot of questions, each version take the form of a `major.minor.patch` version, with the possibility of a metadata prefixed by `+`, and version qualifiers like `-beta`.
+
+If you don't know what is talked about with modules or libraries, please refer yourself to the [<abbr title="Quilt Standard Libraries">QSL</abbr> Structure RFC](../rfc/0009-qsl-structure.md).
+We will also use the word "root library" to design the Maven artifact that depends on every <abbr title="Quilt Standard Libraries">QSL</abbr> libraries.
+
+### Values of `major`, `minor`, and `patch`
+
+When a module or library is created, the `major` number should always be `1` at the start.
+It mays increment in the case of breaking change, whether it is caused by the <abbr title="Quilt Standard Libraries">QSL</abbr> team or by an update to Minecraft.
+
+If a module is updated with a breaking change, the library that includes it should also increment its `major` number.
+Meaning that the root library version should also have its `major` number incremented.
+
+The `minor` number should always start at `0`, and will increment every time a module/library is updated with backwards-compatible changes. It is reset when the `major` number changes.
+
+The `patch` number should always start at `0`, and will increment every time a module/library is updated to fix bugs in a backwards-compatible manner.
+This should not cover the case of a port as [the metadata](#metdata-content) already gives enough information.
```

Comment:
> the metadata already gives enough information.

From a technical perspective, the metadata actually gives **no information**; [see here](https://semver.org/#spec-item-10):
> Build metadata MUST be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence.

In other words, if Loader is given `0.1.0+21w11a` and `0.1.0+21w12a`, and both mods are compatible with the current Minecraft version, it is random which will be loaded.

Reply:
> It is defined which one will be loaded using the depends section, if one depends on 21w22a, and the other one depend on 21w11a strictly, and we're loading 21w22a, then the 21w22a version will be picked up.

Then direct in this RFC for QSL to always depend exactly on 1 version.

> Also doesn't semver define lexicographical order for metadata?

I quoted the section where it explicitly excludes metadata from that in my first comment.

Reply:
Gotta have my eyes checked I guess.

Unresolved question found! 
Is depending on exactly 1 version desirable?